cluster/fpindexcluster/server.py

```diff
@@ -120,6 +121,75 @@ async def get_index_status(request):
         return json_response({}, status=404)
 
 
+async def update_index(request: Request):
+    """Handle fingerprint updates - POST /{index}/_update"""
+    index_name = request.match_info["index"]
+    manager = request.app["index_manager"]
+
+    # Validate index name
+    if not is_valid_index_name(index_name):
+        logger.warning(f"Invalid index name: '{index_name}'")
+        return json_response(
+            {
+                "error": "Invalid index name. Must start with alphanumeric character and contain only alphanumeric, underscore, and hyphen characters.",
+                "index": index_name,
+            },
+            status=400,
+        )
+
+    try:
+        # Parse request body as JSON (following fpindex UpdateRequestJSON structure)
+        request_data = await request.json()
```

Comment:
Let msgspec parse this, it will do validation. You just need to check that each change has either insert or delete not None.